src/hotspot/share/interpreter/rewriter.cpp

```diff
@@ -185,11 +185,11 @@ void Rewriter::rewrite_field_reference(address bcp, int offset, bool reverse) {
   if (!reverse) {
     int cp_index = Bytes::get_Java_u2(p);
     int field_entry_index = _cp_map.at(cp_index);
-    Bytes::put_native_u2(p, field_entry_index);
+    Bytes::put_native_u2(p, checked_cast<u2>(field_entry_index));
```

Comment:
I would have put the checked_cast on the initialization and not the use, but this is OK too.